tests/Unit/PointwiseFunctions/Hydro/Test_StressEnergy.cpp

```diff
@@ -7,27 +7,195 @@
+  }
+  get(spatial_velocity_squared) = 9. / 25.;
+
+  dot_product(make_not_null(&magnetic_field_squared), magnetic_field,
+              magnetic_field, spatial_metric);
+  dot_product(make_not_null(&magnetic_field_dot_spatial_velocity),
+              magnetic_field, spatial_velocity, spatial_metric);
+  hydro::lorentz_factor(make_not_null(&lorentz_factor_v),
+                        spatial_velocity_squared);
+  get(one_over_w_squared) = 1. / square(get(lorentz_factor_v));
+
+  tenex::evaluate<ti::i>(
+      make_not_null(&spatial_velocity_one_form),
+      spatial_velocity(ti::J) * spatial_metric(ti::j, ti::i));
+
+  tenex::evaluate<ti::i>(make_not_null(&magnetic_field_one_form),
+                         magnetic_field(ti::J) * spatial_metric(ti::j, ti::i));
+
+  hydro::comoving_magnetic_field_squared(
+      make_not_null(&comoving_magnetic_field_magnitude_v),
+      magnetic_field_squared, magnetic_field_dot_spatial_velocity,
+      lorentz_factor_v);
+  get(comoving_magnetic_field_magnitude_v) =
+      sqrt(get(comoving_magnetic_field_magnitude_v));
+
+  hydro::stress_energy_tensor(
+      make_not_null(&stress_energy_tensor_v), rest_mass_density,
+      specific_internal_energy, pressure, lorentz_factor_v, lapse,
+      comoving_magnetic_field_magnitude_v, spatial_velocity, shift,
+      magnetic_field, spatial_metric, inverse_spatial_metric);
+
+  gr::spacetime_metric(make_not_null(&spacetime_metric_v), lapse, shift,
+                       spatial_metric);
+
+  gr::inverse_spacetime_metric(make_not_null(&inverse_spacetime_metric_v),
+                               lapse, shift, inverse_spatial_metric);
+
+  tnsr::a<DataVector, 3> four_velocity_one_form_buffer;
+  tnsr::a<DataVector, 3> comoving_magnetic_field_one_form_buffer;
+  ::grmhd::GhValenciaDivClean::trace_reversed_stress_energy(
+      make_not_null(&trace_reversed_stress_energy_tensor_v),
+      make_not_null(&four_velocity_one_form_buffer),
+      make_not_null(&comoving_magnetic_field_one_form_buffer),
+      rest_mass_density, spatial_velocity_one_form, magnetic_field_one_form,
+      magnetic_field_squared, magnetic_field_dot_spatial_velocity,
+      lorentz_factor_v, one_over_w_squared, pressure, specific_internal_energy,
+      spacetime_metric_v, shift, lapse);
+
+  tenex::evaluate<ti::a, ti::b>(make_not_null(&stress_energy_tensor_lowered),
+                                stress_energy_tensor_v(ti::C, ti::D) *
+                                    spacetime_metric_v(ti::c, ti::a) *
+                                    spacetime_metric_v(ti::d, ti::b));
+  tenex::evaluate<ti::a, ti::b>(
+      make_not_null(&trace_reversed_stress_energy_tensor_calc),
+      stress_energy_tensor_lowered(ti::a, ti::b) -
+          0.5 * spacetime_metric_v(ti::a, ti::b) *
+              inverse_spacetime_metric_v(ti::C, ti::D) *
+              stress_energy_tensor_lowered(ti::d, ti::c));
+
+  Approx approx = Approx::custom().epsilon(1e-10);
```

Comment:
`1.e-10` (see second last bullet here https://spectre-code.org/code_review_guide.html )

Reply:
okay. changed it.